app/controllers/review_mapping_controller.rb

```diff
@@ -51,129 +47,196 @@ def select_metareviewer
     @mapping = ResponseMap.find(params[:id])
   end
 
+
+  #EDITED: If the user is invalid(not present in database), then return with error message.
+  #If reviewer is a participant of the assignment, then only move forward with reviewer assignment.
+  #Work Done: Refactored add_reviewer and extracted some code to make it smaller.
   def add_reviewer
-    assignment = Assignment.find(params[:id])
+    assignment = find_assignment(params[:id])
     topic_id = params[:topic_id]
-    user_id = User.where(name: params[:user][:name]).first.id
-    # If instructor want to assign one student to review his/her own artifact,
+    user_id = if User.where(name: params[:user][:name]).first
+      then User.where(name: params[:user][:name]).first.id
+      else nil
+      end
+    if user_id.nil?
+      flash[:error] = 'The user does not exist.'
+      redirect_to action: 'list_mappings', id: assignment.id, msg: 'The user does not exist.'
+      return
+    end
+    # If instructor wants to assign one student to review his/her own artifact,
     # it should be counted as "self-review" and we need to make /app/views/submitted_content/_selfreview.html.erb work.
```

Comment:
This comment appears to say that the instructor can assign a self-review, but the code says he can't.  Please explain.